src/mscorlib/src/System/Diagnostics/Stacktrace.cs

```diff
@@ -618,5 +636,58 @@ private static bool ShowInStackTrace(MethodBase mb)
             Debug.Assert(mb != null);
             return !(mb.IsDefined(typeof(StackTraceHiddenAttribute)) || (mb.DeclaringType?.IsDefined(typeof(StackTraceHiddenAttribute)) ?? false));
         }
+
+        private static bool TryResolveStateMachineMethod(ref MethodBase method, out Type declaringType)
+        {
+            Debug.Assert(method != null);
+            Debug.Assert(method.DeclaringType != null);
+
+            declaringType = method.DeclaringType;
+
+            Type parentType = declaringType.DeclaringType;
+            if (parentType == null)
+            {
+                return false;
+            }
+
+            MethodInfo[] methods = parentType.GetMethods(BindingFlags.Public | BindingFlags.NonPublic | BindingFlags.Static | BindingFlags.Instance | BindingFlags.DeclaredOnly);
+            if (methods == null)
+            {
+                return false;
+            }
+
+            foreach (MethodInfo candidateMethod in methods)
+            {
+                IEnumerable<StateMachineAttribute> attributes;
+                try
+                {
+                    attributes = candidateMethod.GetCustomAttributes<StateMachineAttribute>();
+                    if (attributes == null)
+                    {
+                        continue;
+                    }
+                }
+                catch
+                {
+                    // Linq.Expression compiled functions can throw when checking attributes
```

Comment:
What does it throw? Is that a bug?

Reply:
Don't think need this; dynamic expressions have no declaring type so it wouldn't get here; and the other issues (https://github.com/benaadams/Ben.Demystifier/issues/31, https://github.com/benaadams/Ben.Demystifier/issues/20) appear to be full framework bugs